### PR TITLE
IBC wrapper

### PR DESCRIPTION
**Pull request by timlind**

### CODE REVIEW

File: x/ibctransfer/whitelisting.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+		// The denomination used to send the coins is either the native denom or the hash of the path
+		// if the denomination is not native.
+		denomTrace := transfertypes.ParseDenomTrace(unprefixedDenom)
+		if denomTrace.Path != "" {
+			denom = denomTrace.IBCDenom()
+		}
+
+		return denom
+	}
```
Comment (Jedi2002) [inline x/ibctransfer/whitelisting.go:55]: Cant we just return denom,bool . 
transfertypes.ReceiverChainIsSource is true "",true
transfertypes.ReceiverChainIsSource is false denom,false

And check for whitelist only in the second case .

Comment (Jedi2002) [reply x/ibctransfer/whitelisting.go:55]: Assuming that we can allow anything that went from sifchain to come back

Comment (Jedi2002) [reply x/ibctransfer/whitelisting.go:55]: Oh are we planning to keep non 18 decimal denoms in the whitelist too ? And use the same whitelist to get their units ?

File: x/ibctransfer/whitelisting_test.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	transfertypes "github.com/cosmos/cosmos-sdk/x/ibc/applications/transfer/types"
+	channeltypes "github.com/cosmos/cosmos-sdk/x/ibc/core/04-channel/types"
+	"github.com/stretchr/testify/require"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+func TestIsRecvPacketAllowed(t *testing.T) {
```
Comment (iverc) [inline x/ibctransfer/whitelisting_test.go:13]: I think we could have more tests to cover all whitelisting functions, ideally success/failure case for each one

**Review by iverc — CHANGES_REQUESTED** [review]

**Review by Jedi2002 — APPROVED** [review]

File: x/ibctransfer/whitelisting_test.go (hunk 2 above)
Comment (timlind) [reply x/ibctransfer/whitelisting_test.go:13]: Yup, will do once we integrate the whitelist keeper.

**Pull request merged**